pjmedia/src/pjmedia/stream.c

```diff
@@ -1888,64 +1890,22 @@ static void on_rx_rtp( pjmedia_tp_cb_param *param)
     if (channel->paused)
 	goto on_return;
 
-    /* Update RTP session (also checks if RTP session can accept
-     * the incoming packet.
-     */
-    check_pt = (hdr->pt != stream->rx_event_pt) && PJMEDIA_STREAM_CHECK_RTP_PT;
-    pjmedia_rtp_session_update2(&channel->rtp, hdr, &seq_st, check_pt);
-#if !PJMEDIA_STREAM_CHECK_RTP_PT 
-    if (!check_pt && hdr->pt != channel->rtp.out_pt &&
-	hdr->pt != stream->rx_event_pt) 
-    { 
-	seq_st.status.flag.badpt = 1; 
-    } 
-#endif 
-    if (seq_st.status.value) {
-	TRC_  ((stream->port.info.name.ptr,
-		"RTP status: badpt=%d, badssrc=%d, dup=%d, "
-		"outorder=%d, probation=%d, restart=%d",
-		seq_st.status.flag.badpt,
-		seq_st.status.flag.badssrc,
-		seq_st.status.flag.dup,
-		seq_st.status.flag.outorder,
-		seq_st.status.flag.probation,
-		seq_st.status.flag.restart));
-
-	if (seq_st.status.flag.badpt) {
-	    PJ_LOG(4,(stream->port.info.name.ptr,
-		      "Bad RTP pt %d (expecting %d)",
-		      hdr->pt, channel->rtp.out_pt));
-	}
-
-	if (!stream->si.has_rem_ssrc && seq_st.status.flag.badssrc) {
-	    PJ_LOG(4,(stream->port.info.name.ptr,
-		      "Changed RTP peer SSRC %d (previously %d)",
-		      channel->rtp.peer_ssrc, stream->rtcp.peer_ssrc));
-	    stream->rtcp.peer_ssrc = channel->rtp.peer_ssrc;
-	}
-
+    /* Check SSRC. */
+    if (!channel->rtp.has_peer_ssrc && channel->rtp.peer_ssrc == 0)
+        channel->rtp.peer_ssrc = pj_ntohl(hdr->ssrc);
 
-    }
-
-    /* Skip bad RTP packet */
-    if (seq_st.status.flag.bad) {
-	pkt_discarded = PJ_TRUE;
-	goto on_return;
-    }
-
-    /* Ignore if payloadlen is zero */
-    if (payloadlen == 0) {
-	pkt_discarded = PJ_TRUE;
-	goto on_return;
+    if (pj_ntohl(hdr->ssrc) != channel->rtp.peer_ssrc) {
+	flag_badssrc = 1;
+	if (!channel->rtp.has_peer_ssrc)
+	    channel->rtp.peer_ssrc = pj_ntohl(hdr->ssrc);
```

Comment:
looks like `channel->rtp` is manually updated by app (also in previous change block), is this really necessary?

Reply:
It was taken from [here](https://github.com/pjsip/pjproject/blob/master/pjmedia/src/pjmedia/rtp.c#L248).
Set the peer_ssrc : 
- If remote doesn't specify its ssrc and peer_ssrc is not yet set
- If remote doesn't specify its ssrc and peer_ssrc is set however diferent than the one received, allow the ssrc to be changed.

Reply:
Yes, aware of where it was taken from actually. But, is it necessary or what for?

Reply:
Trying to rephrase here, if app needs to assign `flag_badssrc` (btw, is it _bad_ or just _unrecognized_?), it is okay. But unless it is really necessary, don't update RTP session too (`channel->rtp`) as it should be done via `pjmedia_rtp_session_update2()` which is called later.